packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -446,7 +446,25 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
         context,
         _isChildExpanded(index),
       );
-      final Row header = Row(
+      // ExpandIcon needs to be wrapped with MergeSemantics. If canTapOnHeader
+      // is false, it is possible for headerWidget to contain semantics
+      // boundary. In that case, we wrap the ExpandIcon in expandIconContainer
+      // level. Otherwise, we wrap the entire header as a whole.
+      Widget expandIconContainer = Container(
+        margin: const EdgeInsetsDirectional.only(end: 8.0),
+        child: ExpandIcon(
+          isExpanded: _isChildExpanded(index),
+          padding: const EdgeInsets.all(16.0),
+          onPressed: !child.canTapOnHeader
+              ? (bool isExpanded) => _handlePressed(isExpanded, index)
+              : null,
+        ),
+      );
+      expandIconContainer = !child.canTapOnHeader
```

Comment:
The formatting here is off. This may also be easier to read as:

if (!child.canTapOnHeader) {
 expandIconContainer = MergeSemantics(...);
}